lib/core/src/server/presets.js

```diff
@@ -0,0 +1,66 @@
+import path from 'path';
+import { logger } from '@storybook/node-logger';
+import serverRequire from './serverRequire';
+
+function loadPreset(preset) {
+  try {
+    if (typeof preset === 'string') {
+      return {
+        // eslint-disable-next-line global-require,import/no-dynamic-require
+        preset: require(preset),
+        options: {},
+      };
+    }
+
+    const { name, options } = preset;
+
+    return {
+      // eslint-disable-next-line global-require,import/no-dynamic-require
+      preset: require(name),
+      options,
+    };
+  } catch (e) {
+    logger.warn(`  Failed to load preset: ${JSON.stringify(preset)}`);
+    return false;
+  }
+}
+
+function loadPresets(presets) {
+  if (!presets || !Array.isArray(presets) || !presets.length) {
+    return [];
+  }
+
+  logger.info('Loading presets');
+
+  return presets.map(loadPreset).filter(preset => preset);
+}
+
+function applyPresets(presets, config, extension) {
+  if (!presets.length) {
+    return config;
+  }
+
+  return presets.reduce((accumulatedConfig, { preset, options }) => {
+    const extensionFn = preset[extension];
+
+    if (extensionFn && typeof extensionFn === 'function') {
+      return extensionFn.call(preset, accumulatedConfig, options);
```

Comment:
I think performance-wise it's not important, and api-wise it would look like this:
```js
return extensionFn.apply(preset, [accumulatedConfig, options]);
```

Reply:
I think `apply` would allow `this` to be binded which would be better readable.

Reply:
Oh. Sorry, I got confused here. You are right.

Reply:
👌